app/src/lib/ui/OrientationManager/OrientationForm.svelte

```diff
@@ -14,67 +11,103 @@
 	import LoaderIndicator from '$lib/ui/utils/LoaderIndicator.svelte';
-	import { openComponent } from '$lib/stores';
+	import { connectedUser, openComponent } from '$lib/stores';
 	import { displayFullName } from '../format';
+	import { getOrientationSystemLabel } from '$lib/utils/getOrientationSystemLabel';
 
 	export let displayError = false;
 	export let formTitle = 'Réorienter';
 	export let handleSubmit: (values: OrientationValidationSchema) => Promise<void>;
 
 	export let structureId: string = null;
 	let selectedStructureId = structureId;
+	let selectedOrientationSystemId: string = null;
 
-	const orientationTypes: OperationStore<GetOrientationTypeQuery> = operationStore(
-		GetOrientationTypeDocument
+	const deploymentId = $connectedUser.deploymentId;
+
+	const getProfessionals: OperationStore<GetProfessionalsForDeploymentQuery> = operationStore(
+		GetProfessionalsForDeploymentDocument,
+		{
+			deploymentId: deploymentId,
+		}
 	);
-	query(orientationTypes);
 
-	$: orientationOptions =
-		$orientationTypes.data?.orientation_type.map(({ id, label }) => ({
-			name: id,
-			label,
-		})) ?? [];
+	query(getProfessionals);
 
-	const structures: OperationStore<GetStructuresWithProQuery> = operationStore(
-		GetStructuresWithProDocument,
-		null,
-		{ requestPolicy: 'network-only' }
-	);
-	query(structures);
-	$: structureOptions =
-		$structures.data?.structure.map(({ id, name, professionals }) => {
-			const beneficiaryCount = professionals.reduce(
-				(total: number, value: GetStructuresWithProQuery['structure'][0]['professionals'][0]) => {
-					return total + value.account.referentCount.aggregate.count;
-				},
-				0
-			);
-			return {
-				name: id,
-				label: `${name} (${beneficiaryCount})`,
-			};
-		}) ?? [];
+	$: orientationSystemsOptions =
+		$getProfessionals.data?.professional
+			.flatMap(({ orientationSystems }) => {
+				return orientationSystems.map(({ orientationSystem }) => {
+					return {
+						name: orientationSystem.id,
+						label: getOrientationSystemLabel(orientationSystem),
+					};
+				});
+			})
+			// Sort by alphabetical order
+			.sort((a, b) => a.label.localeCompare(b.label))
+			// Get unique values
+			.filter((value, index, self) => index === self.findIndex((t) => t.name === value.name)) ?? [];
 
-	$: structure = $structures.data?.structure.find(({ id }) => id === selectedStructureId) ?? null;
+	$: structureOptions =
+		$getProfessionals.data?.professional
+			// Get professionnals that have the selected orientation system attached
+			.filter(({ orientationSystems }) =>
+				orientationSystems
+					.map(({ orientationSystemId }) => orientationSystemId)
+					.includes(selectedOrientationSystemId)
+			)
+			// Get the structures of this professional
+			.map(({ structure }) => {
+				const beneficiaryCount = structure.professionals.reduce(
+					(
+						total: number,
+						value: GetProfessionalsForDeploymentQuery['professional'][0]['structure']['professionals'][0]
```

Comment:
Je viens d'apprendre que c'était aussi possible d'écrire `[number]` plutôt que `[0]` et je trouve ça un peu plus élégant :
```suggestion
						value: GetProfessionalsForDeploymentQuery['professional'][number]['structure']['professionals'][number]
```